docs/transparent-proxy.md

```diff
@@ -0,0 +1,268 @@
+DHCP) through and does not have an IP address in the source/target networks, making it almost invisible to the hosts in
+the network. Only traffic destined for the target server on port 3389 is ever processed at the IP level and redirected
+to the MITM through `ebtables`.
+
+
+**Netplan**
+```yaml
+network:
+  version: 2
+  renderer: networkd
+  ethernets:
+    enp0s3:
+      dhcp4: no
+    enp0s8:
+      dhcp4: no
+    # management interface
+    enp0s9:
+      dhcp4: yes
+  bridges:
+    br0:
+      interfaces:
+        - enp0s3
+        - enp0s8
+      # if you are confident that your deployment is not going
+      # to cause a loop you should leave these enabled, otherwise comment them
+      parameters:
+        stp: false
+        forward-delay: 0
+```
+
+**setup.sh**
+
+```shell
+#!/bin/bash -x
+#               enp0s3            enp0s8
+#  +--------+         \ +------+ /           +--------+
+#  | CLIENT | <-- 1 --> | MITM | <--- 2 ---> | SERVER |
+#  +--------+           +------+             +--------+
+#  10.13.37.10             br0              10.13.37.111
+
+# The IP of the RDP server which will receive proxied connections.
+SERVER_IP=10.13.37.111
+
+# The mark number to use in iptables (should be fine as-is)
+MARK=1
+
+# The routing table ID for custom rules (should be fine as-is)
+TABLE_ID=100
+
+# Create a custom routing table for pyrdp traffic
+echo "$TABLE_ID    pyrdp" >> /etc/iproute2/rt_tables
+
+# Intercept RDP traffic destined for the server (1)
+iptables -t mangle -I PREROUTING -p tcp -d 10.13.37.111 --dport 3389 \
+  -j TPROXY --tproxy-mark 0x1/0x1 --on-port 3389 --on-ip 127.0.0.1
+
+# Mark RDP return traffic intended for clients (2)
+iptables -t mangle -A PREROUTING \
+    -s $SERVER_IP \
+    -m tcp -p tcp --sport 3389 \
+    -j MARK --set-mark $MARK
+
+# Set route lookup to the pyrdp table for marked packets.
+ip rule add fwmark $MARK lookup $TABLE_ID
+
+# Add a custom route that redirects traffic intended for the outside world to loopback
+# So that server-client traffic passes through PyRDP
+# This table will only ever be used by RDP so it should not be problematic
+ip route add local default dev lo table $TABLE_ID
+
+# If you want the interception to happen at the Layer 2
+# Make sure that both your interfaces are bridged together
+
+# WARNING: if you have other important firewall rules, make sure to test the impact of this
+modprobe br_netfilter
+echo 1 > /proc/sys/net/bridge/bridge-nf-call-iptables
+
+# Target DROP in brouting means out of L2 go route in L3
+ebtables -t broute -A BROUTING -i enp0s3 -p ipv4 --ip-dst $SERVER_IP --ip-proto tcp --ip-dport 3389 -j redirect --redirect-target DROP
+ebtables -t broute -A BROUTING -i enp0s8 -p ipv4 --ip-src $SERVER_IP --ip-proto tcp --ip-source-port 3389 -j redirect --redirect-target DROP
+# recent linux like debian buster
+#ebtables-legacy -t broute -A BROUTING -i enp0s8 -p ipv4 --ip-src $SERVER_IP --ip-proto tcp --ip-source-port 3389 -j redirect --redirect-target DROP
```

Comment:
just point to the existence of ebtables-legacy instead of a whole copy of the statement